.travis.yml

```diff
@@ -40,6 +40,30 @@ script:
   - flake8 --select=F --ignore=F401,F405,F403,F811,F821,F841
   - git diff --exit-code
 
+  # New format check: every 'glob' must include 'lsort' to avoid non-determinism.
+  #
+  # Explanation:
+  #   glob by itself returns a list in random order. For determinate order,
+  #   precede the glob with an lsort, as recommended by TCL 'glob' man page.
+  #
+  # Example: given a directory with files 'cells-lvt' and 'cells',
+  #     [glob cells*]        => { cells cells-lvt } ?OR? { cells-lvt cells }
+  #     [lsort [glob cells*] => { cells cells-lvt } ALWAYS
```

Comment:
> Great! Can you add a little more detail to this comment about what kind of problems happen?

@ctorng how's this?

Reply:
[See latest code for final slight change to the comment...]

Reply:
I mean maybe about what actually goes wrong?

Reply:
Hm okay I've made two more attempts at being more specific about what can go wrong. The first one is *really* long winded; it could still go in-line as a comment if you like, and/or I could file it as an issue and refer to the issue in the comment. The second write-up is more concise, but I don't know if it has the specificity that you're after...and/or maybe I'm still on the entirely wrong track...let me know...

------------------------

LONG-WINDED EXAMPLE:

A more rigorous example of what can go wrong can be found in the story of the `read_design` script (ref. mflowgen issues 65, 92, pull 93, commit beb5f9ea7776b9aa58ca0c8aa79944ec9eb5eb2c).

The original `read_design.tcl` code loaded libraries using a glob command
```
 set_attr library [join "
                       [glob -nocomplain inputs/adk/*.lib]
                     "]
```
..where e.g. if the view was "multivt", then this would expand to
```
 set_attr library    [join "
                        adk/stdcells.lib
                        adk/stdcells-bc.lib
                        ...
                     "]
```
..except that sometimes, because glob has no guaranteed order, the list would come out differently e.g. with bc first:
```
 set_attr library    [join "
                        adk/stdcells-bc.lib
                        adk/stdcells.lib
                        ...
                     "]
```

..so that a given step asking for a given cell might get the typical-case cell or it might get the best-case cell. E.g. the cell "INVD4BWP16P90" exists in both libraries stdcell and stdcell-bc but with different characteristics. Scripts looking for a given cell generally search the libraries order and stop at the first cell whose name matches.

One place where this manifested as an error is when one step used and specified the cell from the typical library "stdcells.lib"
```
  set_driving_cell -no_design_rule -lib_cell INVD4BWP16P90
     -library tcbn16ffcllbwp16p90tt0p8v25c ...
```

..but than a later step would sometimes fail when trying to load the cell, in those cases when the library load order came out with stdcell-bc in the list before stdcell.lib. The step would find the correct cell name, but the script would fail because the libraries didn't match.
```
     Error   : 'The set_driving_cell' command cannot find a lib_cell named
            'INVD4BWP16P90' in library 'library' specified with the '-libcell' or '-cell' option.
```
Consistent lsort+glob would result in the build always failing or always succeeding, but in this case, glob without lsrot, the build would succeed sometimes but then later fail for seemingly no reason.

------------------------------------------------------------------------
SHORTER VERSION:

As a further example of what can go wrong, a "multivt" view might have two libraries "stdcells-tt.lib" and "stdcells-bc.lib" such that the same cell name appears in both libraries but with different characteristics. If the designer loads the libraries using e.g.
```
  set lib_list [glob stdcells*.lib]
```
they might sometimes get the typical-case cell and other times get the worst-case cell, leading to erratic and unexpected behavior.

Reply:
I like the shorter version!

Reply:
> I like the shorter version!

Me too. I'll stick it in there and then finalize the pull. Thanks for your help!